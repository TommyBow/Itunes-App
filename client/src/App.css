/* App.css */

.header2,
.par {
  margin: 20px 0;
  text-align: center;
}

.header2 {
  font-size: x-large;
  text-decoration: underline;
}

.navbar {
  background-color: #353535;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
  height: 50px;
}

.navbar-brand {
  font-size: 24px;
  color: #f2f2f2;
  font-weight: bold;
  text-align: center;
  margin: 0;
}

.search-results-container,
.favorites-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.search-results-row,
.favorites-row {
  width: 100%;
  display: flex;
  justify-content: space-between;
  margin-bottom: 10px;
}

.search-result-item,
.favorite-item {
  width: calc(25% - 10px);
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-right: 10px;
  margin-bottom: 10px;
}

.search-result-item:last-child,
.favorite-item:last-child {
  margin-right: 0;
}

.search-result-item img,
.favorite-item img {
  width: 100px;
  height: 100px;
  object-fit: cover;
  margin-bottom: 10px;
}

.search-result-item h3,
.favorite-item h3,
.search-result-item p,
.favorite-item p {
  margin: 0;
  text-align: center;
}

.search-result-item button,
.favorite-item button {
  margin-top: 10px;
  padding: 8px 16px;
  background-color: #4CAF50;
  color: #fff;
  border: none;
  border-radius: 4px;
  font-size: 14px;
  cursor: pointer;
  transition: background-color 0.3s ease;
}

.search-result-item button:hover,
.favorite-item button:hover {
  background-color: #45a049;
}

.search-result-item button:focus,
.favorite-item button:focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(0, 123, 255, 0.5);
}

.search-result-item button:active,
.favorite-item button:active {
  background-color: #a9f8ac;
}
