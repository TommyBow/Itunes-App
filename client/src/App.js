import React, { useState } from 'react';
import SearchBar from './SearchBar';
import SearchResult from './SearchResult';
import './App.css';

const App = () => {
  // State variables for search results and favorites
  const [searchResults, setSearchResults] = useState([]);
  const [favorites, setFavorites] = useState([]);

  // Function to handle the search
  const handleSearch = async (query, mediaType) => {
    try {
      const response = await fetch(`/api/search?query=${query}&mediaType=${mediaType}`);
      if (response.ok) {
        const data = await response.json();
        if (data.resultCount > 0) {
          setSearchResults(data.results);
        } else {
          setSearchResults([]);
        }
      } else {
        setSearchResults([]);
      }
    } catch (error) {
      console.error(error);
      setSearchResults([]);
    }
  };

  // Function to add an item to favorites
  const handleAddToFavorites = (item) => {
    setFavorites((prevFavorites) => [...prevFavorites, item]);
  };

  // Function to remove an item from favorites
  const handleRemoveFromFavorites = (item) => {
    setFavorites((prevFavorites) => prevFavorites.filter((fav) => fav !== item));
  };

  // Calculate the number of rows needed based on the number of search results
  const numRows = Math.ceil(searchResults.length / 4);

  // Generate an array of row indexes
  const rowIndexes = Array.from({ length: numRows }, (_, index) => index);

  return (
    <div>
      <nav className="navbar">
        <h1 className="navbar-brand">iTunes Search App</h1>
      </nav>
      
      <SearchBar handleSearch={handleSearch} />

      <h2 className="header2">Search Results:</h2>
      {searchResults.length > 0 ? (
        <div className="search-results-container">
          {rowIndexes.map((rowIndex) => (
            <div key={rowIndex} className="search-results-row">
              {searchResults
                .slice(rowIndex * 4, rowIndex * 4 + 4)
                .map((result, index) => (
                  <SearchResult
                    key={index}
                    result={result}
                    handleAddToFavorites={handleAddToFavorites}
                  />
                ))}
            </div>
          ))}
        </div>
      ) : (
        <p className="par">No search results found.</p>
      )}

      <h2 className="header2">My Favorites:</h2>
      <div className="search-results-container">
        {rowIndexes.map((rowIndex) => (
          <div key={rowIndex} className="search-results-row">
            {favorites
              .slice(rowIndex * 4, rowIndex * 4 + 4)
              .map((favorite, index) => (
                <div key={index} className="search-result-item">
                  <img src={favorite.artworkUrl100} alt={favorite.trackName} />
                  <h3>{favorite.trackName}</h3>
                  <p>{favorite.artistName}</p>
                  <p>{favorite.kind}</p>
                  <button onClick={() => handleRemoveFromFavorites(favorite)}>Remove from Favorites</button>
                </div>
              ))}
          </div>
        ))}
      </div>
    </div>
  );
};

export default App;