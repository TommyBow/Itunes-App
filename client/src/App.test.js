import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders app correctly', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
