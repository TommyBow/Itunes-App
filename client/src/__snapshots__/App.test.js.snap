// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app correctly 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="header1"
    >
      iTunes Search App
    </h1>
    <form
      class="search-bar"
    >
      <input
        class="search-bar__input"
        placeholder="Enter search term"
        type="text"
        value=""
      />
      <select
        class="search-bar__select"
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="movie"
        >
          Movie
        </option>
        <option
          value="podcast"
        >
          Podcast
        </option>
        <option
          value="music"
        >
          Music
        </option>
        <option
          value="audiobook"
        >
          Audiobook
        </option>
        <option
          value="shortFilm"
        >
          Short Film
        </option>
        <option
          value="tvShow"
        >
          TV Show
        </option>
        <option
          value="software"
        >
          Software
        </option>
        <option
          value="ebook"
        >
          Ebook
        </option>
      </select>
      <button
        class="search-bar__button"
        type="submit"
      >
        Search
      </button>
    </form>
    <h2
      class="header2"
    >
      Search Results
    </h2>
    <p
      class="par"
    >
      No search results found.
    </p>
    <h2
      class="header2"
    >
      Favorites
    </h2>
    <div
      class="search-results-container"
    />
  </div>
</DocumentFragment>
`;
